### PR TITLE
Gradle build for Java SDK

### PR DESCRIPTION
**Pull request by fenghao-chen**

### CODE REVIEW

**Review by jtreminio-dropbox — APPROVED** [review]

File: sdks/java/.github/workflows/github-actions.yml [hunk 1]
```diff
@@ -1,13 +1,67 @@
...
+
+      - name: Retrieve version
+        run: |
+          echo "VERSION_NAME=$(cat gradle.properties | grep -w "VERSION_NAME" | cut -d'=' -f2)" >> $GITHUB_ENV
+      - name: Publish Release to Maven Central
+        run: bin/gradle ./gradlew closeAndReleaseRepository --no-daemon --no-parallel
+        if: "!endsWith(env.VERSION_NAME, '-SNAPSHOT')"
+        env:
+          ORG_GRADLE_PROJECT_mavenCentralUsername: ${{ secrets.SONATYPE_USERNAME }}
+          ORG_GRADLE_PROJECT_mavenCentralPassword: ${{ secrets.SONATYPE_PASSWORD }}
```
Comment (handstandsam) [inline sdks/java/.github/workflows/github-actions.yml:67]: I would comment this all out for now until you have verified that SNAPSHOTs are successfully being uploaded.

File: sdks/java/build.gradle [hunk 2]
```diff
@@ -21,6 +9,21 @@ buildscript {
...
+apply plugin: 'maven'
+apply plugin: 'com.diffplug.spotless'
+apply plugin: 'signing'
+
+group = 'com.dropbox.sign'
+archivesBaseName = 'dropbox-sign'
+version = '1.0.0-beta2301'
+sourceCompatibility = JavaVersion.VERSION_1_8
+targetCompatibility = JavaVersion.VERSION_1_8
+
```
Comment (handstandsam) [inline sdks/java/build.gradle:26]: Set your snapshot host as well to S01

```
mavenPublish {
  sonatypeHost = S01
}
```

https://github.com/dropbox/dependency-guard/blob/d758d06ff46a4b536067b4a379754a08cba8b5d7/dependency-guard/build.gradle.kts#L58-L60

File: sdks/java/.github/workflows/github-actions.yml (hunk 1 above)
Comment (handstandsam) [reply sdks/java/.github/workflows/github-actions.yml:67]: Make sure https://github.com/hellosign/hellosign-openapi/blob/main/sdks/java/gradle.properties#L9 is set to `-SNAPSHOT`

File: sdks/java/build.gradle (hunk 2 above)
Comment (handstandsam) [reply sdks/java/build.gradle:26]: Once the snapshot is published it'll show up at https://s01.oss.sonatype.org/content/repositories/snapshots/com/dropbox/

**Pull request merged**